lib/services/wallet_connect_service.dart

```diff
@@ -0,0 +1,355 @@
+    if (!shouldTriggerEvent()) {
+      return;
+    }
+    web3Wallet!
+        .getActiveSessions()
+        .forEach(((String session, SessionData sessionData) {
+      if (sessionData.namespaces.entries.first.value.events
+          .contains(WcEvents.amountChanged)) {
+        List<dynamic> data = [];
+        for (var id in changedIDs.entries) {
+          dynamic item = {};
+          item["publicId"] = id.key;
+          item["amount"] = id.value;
+          data.add(item);
+        }
+
+        web3Wallet!.emitSessionEvent(
+            topic: sessionData.topic,
+            chainId: Config.walletConnectChainId,
+            event:
+                SessionEventParams(name: WcEvents.amountChanged, data: data));
+      }
+    }));
+  }
+
+//Triggers an token amount change event for the wallet connect clients
+//@param changedIDs Map<String, List<QubicAssetDto>> (key = publicId) , List ,containes changed token amounts
+  void triggerTokenAmountChangedEvent(
+      Map<String, List<QubicAssetDto>> changedIDs) {
+    if (!shouldTriggerEvent()) {
+      return;
+    }
+
+    web3Wallet!
+        .getActiveSessions()
+        .forEach(((String session, SessionData sessionData) {
+      if (sessionData.namespaces.entries.first.value.events
+          .contains(WcEvents.tokenAmountChanged)) {
+        List<dynamic> data = [];
+        for (var id in changedIDs.entries) {
+          dynamic item = {};
+
+          List<dynamic> tokens = [];
+          tokens.addAll(
+              id.value.map((e) => QubicAssetWC.fromQubicAssetDto(e)).toList());
+
+          item["publicId"] = id.key;
+          item["tokens"] = tokens;
+          data.add(item);
+        }
+
+        web3Wallet!.emitSessionEvent(
+            topic: sessionData.topic,
+            chainId: Config.walletConnectChainId,
+            event: SessionEventParams(
+                name: WcEvents.tokenAmountChanged, data: data));
+      }
+    }));
+  }
+
+  //Triggers an accountsChanged event for all the wallet connect clients who have subscribed to it
+  void triggerAccountsChangedEvent() {
+    if (!shouldTriggerEvent()) {
+      return;
+    }
+    web3Wallet!
+        .getActiveSessions()
+        .forEach(((String session, SessionData sessionData) {
+      if (sessionData.namespaces.entries.first.value.events
+          .contains(WcEvents.accountsChanged)) {
+        List<dynamic> data = [];
+        for (var id in appStore.currentQubicIDs) {
+          if (id.watchOnly == false) {
+            dynamic item = {};
+            item["publicId"] = id.publicId;
+            item["name"] = id.name;
+            item["amount"] = id.amount ?? -1;
+            data.add(item);
+          }
+        }
+
+        web3Wallet!.emitSessionEvent(
+            topic: sessionData.topic,
+            chainId: Config.walletConnectChainId,
+            event:
+                SessionEventParams(name: WcEvents.accountsChanged, data: data));
+      }
```

Comment:
even when the data supports any object, it seems to me that DApps will be expecting an array of strings, where each string ins the address in CAIP10 format.
Then, we might have a separated event that returns more details of the account if needed. However, I think maybe only the name/alias would be 'needed' (while the qubic namespace service is not ready :) )

Reply:
Actually it's totally up to us, since there's no standard way on how dApp should expect account info (CAIP10 is for Wallet Connect, not the dApp side). I am thinking that providing all the required info would allow for greater flexibility of dApp developers. But it's your call to return CAIP10 info only or full object